HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -17,6 +17,24 @@
 #                     pset.minGoodStripCharge = cms.PSet(refToPSet_ = cms.string('HLTSiStripClusterChargeCutNone'))
 #     return process
 
+# Eta Extended Electrons 
+def customiseFor35309(process):
+    for pset in process._Process__psets.values():
+        if hasattr(pset,'ComponentType'):
+            if (pset.ComponentType == 'CkfBaseTrajectoryFilter'):
+                if not hasattr(pset, 'highEtaSwitch'):
+                    pset.highEtaSwitch = cms.double(5.0)
+                if not hasattr(pset, 'minHitsAtHighEta'):
+                    pset.minHitsAtHighEta = cms.int32(5)
+
+    for esp in esproducers_by_type(process, 'KFFittingSmootherESProducer'):
+        if not hasattr(esp, 'HighEtaSwitch'):
+            esp.HighEtaSwitch = cms.double(5.0)
+        if not hasattr(esp, 'MinNumberOfHitsHighEta'):
+            esp.MinNumberOfHitsHighEta = cms.int32(5)
+
+    return process
```

Comment:
If these added parameters are all in their respective fillDescriptions method, then this customisation would not be needed. Is this the case or could it be achieved?

Reply:
It doesn't seem straightforward to me, as fillDescriptions function is not present in one of the codes where we added new parameter.

Reply:
Which one? Could it be added there?

Reply:
so this part can be deleted:
```
for esp in esproducers_by_type(process, 'KFFittingSmootherESProducer'):
        if not hasattr(esp, 'HighEtaSwitch'):
            esp.HighEtaSwitch = cms.double(5.0)
        if not hasattr(esp, 'MinNumberOfHitsHighEta'):
            esp.MinNumberOfHitsHighEta = cms.int32(5)
```
because `TrackingTools/TrackFitters/plugins/KFFittingSmootherESProducer.cc` already have `fillDescriptions`, and our new parameters are in there.

But the other part, the following, is needed:
```
    for pset in process._Process__psets.values():
        if hasattr(pset,'ComponentType'):
            if (pset.ComponentType == 'CkfBaseTrajectoryFilter'):
                if not hasattr(pset, 'highEtaSwitch'):
                    pset.highEtaSwitch = cms.double(5.0)
                if not hasattr(pset, 'minHitsAtHighEta'):
                    pset.minHitsAtHighEta = cms.int32(5)
```
because new parameters are added in `TrackingTools/TrajectoryFiltering/interface/MinHitsTrajectoryFilter.h`, and there is no `fillDescriptions`. Adding `fillDescriptions` in MinHitsTrajectoryFilter.cc doesn't solve the problem. It's not clear where we need to add `fillDescriptions` to make it work. 

If possible, we would like to proceed with HLT customisation for now, and we will try to figure out `fillDescriptions` later on together with tracking experts.

Reply:
I've now removed `KFFittingSmootherESProducer` customisation and resolved the conflict...

> My bad on the customisation @swagata87; when we discussed it offline, I missed that it was not needed for one of the modules.

oh no problem!

Reply:
> It's not clear where we need to add `fillDescriptions` to make it work.

From a quick look, it seems that the issue is the lack of a `fillDescriptions` in [CkfTrackCandidateMaker](https://github.com/cms-sw/cmssw/blob/8cb2dd13baf320139222d328e95a51bd2bece997/RecoTracker/CkfPattern/plugins/CkfTrackCandidateMaker.h) (which in turn would probably be implemented as a function in [CkfTrackCandidateMakerBase](https://github.com/cms-sw/cmssw/blob/8cb2dd13baf320139222d328e95a51bd2bece997/RecoTracker/CkfPattern/src/CkfTrackCandidateMakerBase.cc)). That being said, this looks like something beyond the scope of this PR.